velox/type/DecimalUtil.h

```diff
@@ -250,6 +250,21 @@ class DecimalUtil {
     return (unsignedSum >> 127);
   }
 
+  /// Adds two int128_t inputs, calculate the sum, and return the overflow.
+  /// It can be used to track the number of overflow when adding a batch of
+  /// input numbers. It takes lhs and rhs as input, and stores their sum in
+  /// result. overflow == 1 indicates upward overflow. overflow == -1
+  /// indicates downward overflow. overflow == 0 indicates no overflow.
+  /// When only one input is negative, adding them will never result in
+  /// overflow, so we can directly add them. When both inputs are positive
+  /// or negative, adding them may result in overflow. We can first convert
+  /// the inputs to uint_128_t, and use uint128_t to store the result of the
+  /// addition. The highest bit of the result can then be used to indicate
+  /// overflow. Based on the signs of the inputs, we can adjust the signs of
+  /// sum and overflow accordingly.
+  /// After following the above logic to sum a batch of input numbers, if the
+  /// final accumulated overflow and sum meet certain conditions, we should
+  /// use the adjustSumForOverflow method below to adjust the result.
   inline static int64_t
   addWithOverflow(int128_t& result, const int128_t& lhs, const int128_t& rhs) {
```

Comment:
I wonder why did we decide to put result in an output parameter and return overflow. It seems more natural to return the result and update 'overflow' output parameter, no?

Reply:
I'm not quite sure if the original author wrote it this way for a specific reason. Your suggested implementation does seem more natural. Should we modify this part in this PR?

Reply:
Let's change that.